CefSharp/IFrame.cs

```diff
@@ -182,5 +182,25 @@ public interface IFrame : IDisposable
         /// <param name="initializePostData">Initialize the PostData object when creating this request</param>
         /// <returns>A new instance of the request</returns>
         IRequest CreateRequest(bool initializePostData = true);
+
+        /// <summary>
+        /// Create a new URL request that will be treated as originating from this frame
+        /// and the associated browser. This request may be intercepted by the client via
+        /// IResourceRequestHandler or ISchemeHandlerFactory.
+        /// Use IURLRequest.Create instead if you do not want the request to have
+        /// this association, in which case it may be handled differently (see documentation on that method).
+        ///
+        /// Requests may originate from both the browser process and the render process.
+        /// For requests originating from the browser process: - POST data may only contain a single element
+        /// of type PDE_TYPE_FILE or PDE_TYPE_BYTES.
+        /// For requests originating from the render process: - POST data may only contain a single element of type PDE_TYPE_BYTES.
+        /// - If the response contains Content-Disposition or Mime-Type header values that would not normally be rendered then
+        /// the response may receive special handling inside the browser
+        /// for example, via the file download code path instead of the URL request code path).
+        ///
+        /// The request object will be marked as read-only after calling this method. 
+        /// </summary>
+        //TODO: Implement IURLRequestClient.
+        IURLRequest CreateURLRequest(IRequest request, IURLRequestClient client);
```

Comment:
Since the request object becomes read-only here due to cef internals, maybe I should add a new ImmutableRequest class with a copy constructor from IRequest and no setters? This would protect the user from making a mistake and crashing cef, but I noticed cefsharp generally doesn't have this level of abstraction/internals protection.

Reply:
> maybe I should add a new ImmutableRequest class with a copy constructor from IRequest and no setters

I'd rather notify the user they are attempting to modify a readonly object rather than silently accepting the change.  I'll add a simple `ThrowIfReadOnly` guard shortly.

Reply:
Also rename to `CreateUrlRequest`

Reply:
`ThrowIfReadonly` guard added in https://github.com/cefsharp/CefSharp/commit/b35d5151fee5113bc2f833665394f1c9c2aa1ff8

Also a readonly copy of the `NameValueCollection` added. Let me know if you think I've missed anything.